### PR TITLE
Announce upcoming breaking changes in packaging

### PR DESCRIPTION
**Pull request by smortex**
In #4043, we will introduce changes that might impact users.  Add a note at installation time to warn them about this future change and how to proceed to not have trouble if they will be affected by the change.

### CODE REVIEW

File: scripts/pkg/build_templates/opensearch-dashboards/deb/debian/postinst [hunk 1]
```diff
@@ -35,6 +35,10 @@ echo " sudo systemctl daemon-reload"
 echo " sudo systemctl enable opensearch-dashboards.service"
 echo "### You can start opensearch-dashboards service by executing"
 echo " sudo systemctl start opensearch-dashboards.service"
+echo "### Upcoming breaking change in packaging"
+echo " We are planing to tighten the permissions of installed files in a future version of the opensearch-dashboards package."
+echo " If you setup tooling that needs read access to the OpenSearch Dashboards configuration files, we invite you to add the user running these tools to the 'opensearch-dashboards' group."
```
Comment (peterzhuamazon) [inline scripts/pkg/build_templates/opensearch-dashboards/deb/debian/postinst:40]: `your` not `you`?

**Comment by peterzhuamazon** [comment]
Hi @smortex,

Thanks for PR.

We need to think about how to only display this msg in 2.x+, not 1.x.
Or have a generic msg that suit for all versions.

cc: @bbarani @gaiksaya .

Thanks.

**Comment by smortex** [comment]
I added a commit that try to make this more clear as you suggested in slack. Maybe native English speaker will be able to help us to make this more straightforward?  The commit message has more details, but I think we deserve something concise as a post-installation message.

File: scripts/pkg/build_templates/opensearch-dashboards/deb/debian/postinst (hunk 1 above)
Comment (smortex) [reply scripts/pkg/build_templates/opensearch-dashboards/deb/debian/postinst:40]: I feel like it's good, but maybe "If you are configuring tools that require read access..." is simpler and better?

**Comment by peterzhuamazon** [comment]
Hi @smortex 

Updated PR with some wording improvements.
Please take one more look before I approve. Thanks.

**Comment by smortex** [comment]
:+1: Excellent! LGTM!

**Review by zelinh — APPROVED** [review]
LGTM.

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**